### DISCUSSION TITLE
When using multiple Middleware, decorator(requires) works with only one Middleware..

### DISCUSSION THREAD

**Discussion by wnesbv** [encode/starlette#2276] [Q&A]
windows 10, python 3.11

`
middleware=[
    Middleware(
        AuthenticationMiddleware,
        backend=JwtBackend(
            key=str(settings.SECRET_KEY),
            algorithm=str(settings.JWT_ALGORITHM),
        ),
    ),
    Middleware(
        PrivilegedMiddleware,
        backend=PrivilegedBackend(
            key=str(settings.SECRET_KEY),
            algorithm=str(settings.JWT_ALGORITHM),
        ),
    ),
]
`

**Comment by Kludex** [comment]
Sorry, I didn't get anything that you said. If you want to be helped, I recommend creating a minimal, reproducible example, and explaining in an easy way.

**Reply by wnesbv** [reply]
https://github.com/wnesbv/starlette-example/blob/main/main.py
https://github.com/wnesbv/starlette-example/blob/main/account/middleware.py
https://github.com/wnesbv/starlette-example/blob/main/auth_privileged/middleware.py
https://github.com/wnesbv/starlette-example/blob/main/templates/auth/details.html
the decorator is triggered by one "Middleware", depending on its order, when reading the script
